lib/ansible/modules/cloud/cloudstack/cs_instance_nic.py

```diff
@@ -0,0 +1,265 @@
+domain:
+  description: Domain the VM is related to.
+  returned: success
+  type: string
+  sample: example domain
+account:
+  description: Account the VM is related to.
+  returned: success
+  type: string
+  sample: example account
+project:
+  description: Name of project the VM is related to.
+  returned: success
+  type: string
+  sample: Production
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (AnsibleCloudStack,
+                                             cs_argument_spec,
+                                             cs_required_together)
+from cs import CloudStackException
+
+
+class AnsibleCloudStackInstanceNic(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackInstanceNic, self).__init__(module)
+        self.nic = None
+        self.returns = {
+            'ipaddress': 'ip_address',
+            'macaddress': 'mac_address',
+            'netmask': 'netmask',
+        }
+
+    def get_nic(self):
+        if self.nic:
+            return self.nic
+        args = {
+            'virtualmachineid': self.get_vm(key='id'),
+            'networkid': self.get_network(key='id'),
+        }
+        nics = self.cs.listNics(**args)
+        if nics:
+            self.nic = nics['nic'][0]
+            return self.nic
+        return None
+
+    def add_nic(self):
+        self.result['changed'] = True
+        args = {
+            'virtualmachineid': self.get_vm(key='id'),
+            'networkid': self.get_network(key='id'),
+        }
+        res = self.cs.addNicToVirtualMachine(**args)
+        if 'errortext' in res:
+            self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+
+        vm = self.poll_job(res, 'virtualmachine')
+        self.nic = vm['nic'][0]
+        return self.nic
+
+    def remove_nic(self):
+        self.result['changed'] = True
+        args = {
+            'virtualmachineid': self.get_vm(key='id'),
+            'nicid': self.get_nic()['id'],
+        }
+        res = self.cs.removeNicFromVirtualMachine(**args)
+        if 'errortext' in res:
+            self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+        self.nic = None
+
+    def present_nic(self):
+        nic = self.get_nic()
+        if nic:
+            return nic
+        return self.add_nic()
+
+    def absent_nic(self):
+        nic = self.get_nic()
+        if nic:
+            self.remove_nic()
+
+    def get_result(self, nic):
+        super(AnsibleCloudStackInstanceNic, self).get_result(nic)
+        if nic and not self.module.params.get('network'):
+            self.module.params['network'] = nic.get('networkid')
+        self.result['network'] = self.get_network(key='name')
+        self.result['vm'] = self.get_vm(key='name')
+        self.result['domain'] = self.get_domain(key='path')
+        self.result['account'] = self.get_account(key='name')
+        self.result['project'] = self.get_project(key='name')
```

Comment:
domain, account, project, are already in the result dict in version 2.4, e.g. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/cloudstack.py#L501